diff --git a/node_modules/phaser-animated-tiles/src/plugin/main.js b/node_modules/phaser-animated-tiles/src/plugin/main.js
index 92cdefc..e60bdcd 100644
--- a/node_modules/phaser-animated-tiles/src/plugin/main.js
+++ b/node_modules/phaser-animated-tiles/src/plugin/main.js
@@ -283,7 +283,7 @@ class AnimatedTiles extends Phaser.Plugins.ScenePlugin {
                             // Go through all layers for tiles
                             map.layers.forEach(
                                 (layer) => {
-                                    if (layer.tilemapLayer.type === "StaticTilemapLayer") {
+                                    if (layer && layer.tilemapLayer && layer.tilemapLayer.type === "StaticTilemapLayer") {
                                         // We just push an empty array if the layer is static (impossible to animate).
                                         // If we just skip the layer, the layer order will be messed up
                                         // when updating animated tiles and things will look awful.
@@ -299,7 +299,7 @@ class AnimatedTiles extends Phaser.Plugins.ScenePlugin {
                                             tileRow.forEach(
                                                 (tile) => {
                                                     // Tiled start index for tiles with 1 but animation with 0. Thus that wierd "-1"
-                                                    if ((tile.index - tileset.firstgid) === index) {
+                                                    if ((tile && tile.index - tileset.firstgid) === index) {
                                                         tiles.push(tile);
                                                     }
                                                 }
@@ -365,7 +365,7 @@ class AnimatedTiles extends Phaser.Plugins.ScenePlugin {
                         tileAnimData.tiles.forEach(
                             (tiles, layerIndex) => {
                                 let layer = mapAnimData.map.layers[layerIndex];
-                                if (layer.type === "StaticTilemapLayer") {
+                                if (!layer.tilemapLayer || !layer.tilemapLayer.type || layer && layer.type === "StaticTilemapLayer") {
                                     return;
                                 }
                                 for (let x = chkX; x < (chkX + chkW); x++) {
